### PR TITLE
create material receipts endpoint && Siro camel route

### PR DESCRIPTION
**Pull request by pvpurcarcosmin**
refs: https://github.com/metasfresh/metasfresh/issues/7186

### CODE REVIEW

**Review by metas-ts — CHANGES_REQUESTED** [review]
lgtm, but please:
* check my comments
* write derialization/deserialization unit tests for the common-shipping json-objects
* write tests for the shipment and receipt route for this you can
  * check out the the tests for shipment-candidates/receipt-candidates route
  * since those are driven by quarz and thus a bit different from your 2 routes, you can also take a look at e.g. `de.metas.edi.esb.invoicexport.ecosio.EcosioInvoicRouteTest` in the branch `intensive_care_uat`

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/InOutProducer.java [hunk 1]
```diff
@@ -108,28 +111,37 @@
...
 
 	/**
 	 *
 	 * @param result
 	 * @param complete
-	 * @param createReceiptWithDatePromised if <code>false</code> (the default), then a new InOut is created with the current date from {@link Env#getDate(Properties)}. Otherwise it is created with
-	 *            the DatePromised value of the receipt schedule's C_Order. To be used e.g. when doing migration work.
-	 */
-	protected InOutProducer(@NonNull final InOutGenerateResult result, final boolean complete, final boolean createReceiptWithDatePromised)
+	 * @param movementDateRule if {@code NONE} (the default), then a new InOut is created with the current date from {@link Env#getDate(Properties)}.
```
[the remark this answers is not in the document]
Comment (pvpurcarcosmin) [reply backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/InOutProducer.java:129]: good point 😄

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/InOutProducer.java [hunk 2]
```diff
@@ -108,28 +112,34 @@
 	private final Set<Integer> _currentOrderIds = new HashSet<>();
 
 	/**
-	 * Calls {@link #InOutProducer(InOutGenerateResult, boolean, boolean)} with <code>createReceiptWithDatePromised == false</code>.
+	 * Calls {@link #InOutProducer(InOutGenerateResult, boolean, ReceiptMovementDateRule, Map)} with <code> ReceiptMovementDateRule.NONE && externalInfoByScheduleId = null</code>.
```
Comment (metas-ts) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/InOutProducer.java:115]:
```suggestion
	 * Calls {@link #InOutProducer(InOutGenerateResult, boolean, ReceiptMovementDateRule, Map)} with <code> ReceiptMovementDateRule.CURRENT_DATE && externalInfoByScheduleId = null</code>.
```

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/InOutProducer.java [hunk 3]
```diff
@@ -108,28 +112,34 @@
...
 	 *
 	 * @param result
 	 * @param complete
```
Comment (metas-ts) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/InOutProducer.java:118]:
```suggestion
```

File: misc/services/camel/de-metas-camel-shipping/src/main/java/de/metas/camel/shipping/shipment/ShipmentXmlToJsonRouteBuilder.java [hunk 4]
```diff
@@ -36,7 +37,7 @@
 				.to(LOCAL_STORAGE_URL)
 				.streamCaching()
 				.unmarshal(jacksonXMLDataFormat)
-				.process(new ShipmentXmlToJsonProcessor())
+				.process(new ShipmentXmlToJsonProcessor()).id(SHIPMENT_XML_TO_JSON_PROCESSOR)
```
Comment (metas-ts) [inline misc/services/camel/de-metas-camel-shipping/src/main/java/de/metas/camel/shipping/shipment/ShipmentXmlToJsonRouteBuilder.java:40]: hm, i think i wasn't aware of this :thinking: i'll try to keep it in mind

File: misc/services/camel/de-metas-camel-shipping/src/test/resources/de/metas/camel/shipping/shipment/SiroCreateShipment.xml [hunk 5]
```diff
@@ -0,0 +1,208 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<FMPXMLRESULT xmlns="http://www.filemaker.com/fmpxmlresult">
+    <ERRORCODE>0</ERRORCODE>
+    <PRODUCT BUILD="04-29-2019"
+             NAME="FileMaker"
+             VERSION="ProAdvanced 18.0.1"/>
+    <DATABASE DATEFORMAT="D.m.yyyy"
+              LAYOUT="xml"
+              NAME="bodymed_aussendungen.fmp12"
```
Comment (metas-ts) [inline misc/services/camel/de-metas-camel-shipping/src/test/resources/de/metas/camel/shipping/shipment/SiroCreateShipment.xml:9]:
```suggestion
              NAME="test_aussendungen.fmp12"
```

File: misc/services/camel/de-metas-camel-shipping/src/test/resources/de/metas/camel/shipping/shipment/SiroCreateShipment_empty.xml [hunk 6]
```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<FMPXMLRESULT xmlns="http://www.filemaker.com/fmpxmlresult">
+    <ERRORCODE>0</ERRORCODE>
+    <PRODUCT BUILD="04-29-2019"
+             NAME="FileMaker"
+             VERSION="ProAdvanced 18.0.1"/>
+    <DATABASE DATEFORMAT="D.m.yyyy"
+              LAYOUT="xml"
+              NAME="bodymed_aussendungen.fmp12"
```
Comment (metas-ts) [inline misc/services/camel/de-metas-camel-shipping/src/test/resources/de/metas/camel/shipping/shipment/SiroCreateShipment_empty.xml:9]:
```suggestion
              NAME="test_aussendungen.fmp12"
```

File: misc/services/camel/de-metas-camel-shipping/src/test/resources/de/metas/camel/shipping/receipt/SiroCreateReceipts_Empty.xml [hunk 7]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<FMPXMLRESULT xmlns="http://www.filemaker.com/fmpxmlresult">
+    <ERRORCODE>0</ERRORCODE>
+    <PRODUCT BUILD="04-29-2019"
+             NAME="FileMaker"
+             VERSION="ProAdvanced 18.0.1"/>
+    <DATABASE DATEFORMAT="D.m.yyyy"
+              LAYOUT=""
+              NAME="bodymed_wareneingang.fmp12"
```
Comment (metas-ts) [inline misc/services/camel/de-metas-camel-shipping/src/test/resources/de/metas/camel/shipping/receipt/SiroCreateReceipts_Empty.xml:9]:
```suggestion
              NAME="test_wareneingang.fmp12"
```

**Review by metas-ts — COMMENTED** [review]
pls remove the customer string, apart from that it's LGTM!

File: misc/services/camel/de-metas-camel-shipping/src/main/java/de/metas/camel/shipping/shipment/ShipmentXmlToJsonRouteBuilder.java (hunk 4 above)
Comment (pvpurcarcosmin) [reply misc/services/camel/de-metas-camel-shipping/src/main/java/de/metas/camel/shipping/shipment/ShipmentXmlToJsonRouteBuilder.java:40]: used it for testing, basically to be able to add a mock node right after it that facilitates checking what the processor did

File: misc/services/camel/de-metas-camel-shipping/src/test/resources/de/metas/camel/shipping/receipt/SiroCreateReceipts_Empty.xml (hunk 7 above)
Comment (pvpurcarcosmin) [reply misc/services/camel/de-metas-camel-shipping/src/test/resources/de/metas/camel/shipping/receipt/SiroCreateReceipts_Empty.xml:9]: yep, thanks for fixing!

**Pull request merged**